Add me.mitya57.ReText (retext-flatpak)

scx opened the pull request:
[ReText](https://github.com/retext-project/retext) is a simple but powerful text editor for Markdown and reStructuredText.

TingPing commented on me.mitya57.ReText.json:
```diff
@@ -0,0 +1,378 @@
+        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+        "--env=QTWEBENGINE_DISABLE_SANDBOX=1",
+        "--env=QT_XCB_FORCE_SOFTWARE_OPENGL=1",
```
I've not seen other apps require that and sounds like it will kill performance.

scx replied:
Qt Web Engine: stability problems may occur when using nouveau or intel driver
https://github.com/lirios/browser/issues/77
https://lists.freedesktop.org/archives/flatpak/2017-June/000722.html

TingPing commented on io.qt.qtwebkit.BaseApp.json:
```diff
@@ -0,0 +1,28 @@
+{
```
Not sure why this is in the repo, can be removed.

TingPing commented on me.mitya57.ReText.json:
```diff
@@ -0,0 +1,378 @@
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://pypi.python.org/packages/e4/15/4e2e49f64884edbab6f833c6fd3add24d7938f2429aec1f2883e645d4d8f/PyQt5-5.10.1-5.10.1-cp35.cp36.cp37.cp38-abi3-manylinux1_x86_64.whl",
```
Build this from source.

scx replied:
Updated:
https://github.com/scx/flathub/commit/b7594045334868fa38c133a0ad83801148c882d7#diff-8259601d3ba825e42e3782ae49e24c14

TingPing commented on me.mitya57.ReText.json:
```diff
@@ -0,0 +1,378 @@
+                    ]
+                }
+            ],
+            "post-install": [
```
This is massive and complex enough that it should be its own file probably. It also should try to make it upstream.

TingPing replied:
To be clear here again since there seems to be confusion. I mean putting it in a bash file outside of json so you can have sane whitespacing, no escaping of text, syntax highlighting, etc.

TingPing commented on qtwebkit.json:
```diff
@@ -0,0 +1,13 @@
+{
```
Also not sure why this is here, should be removed.

TingPing commented on retext.1:
```diff
@@ -0,0 +1,20 @@
+.TH RETEXT 1
```
1. This should be upstreamed
2. This doesn't belong in the flatpak anyway

scx replied:
This is a source file, used for update translation.

TingPing replied:
To be clear, `flatpak` doesn't expose man files, just remove it as no user will ever see it.

scx commented on me.mitya57.ReText.json:
```diff
@@ -0,0 +1,378 @@
+                        "cp 'ReText/window.py' 'ReText/window.py.bak';",
+                        "cp 'ReText/tab.py' 'ReText/tab.py.bak';",
+                        "sed -r -i 's/^([[:blank:]]*)(menuEdit\\.addAction\\(self\\.(actionWebKit|actionWebEngine)\\))([[:blank:]]*)$/\\1pass #\\2\\4/' 'ReText/window.py';",
+                        "sed -r -i 's/^([[:blank:]]*)(globalSettings\\.(useWebKit|useWebEngine)[[:blank:]]*=[[:blank:]]*)(enable)([[:blank:]]*)$/\\1\\2False\\5/' 'ReText/window.py';",
```
I would like to avoid creating a new patch with every new release.
Moreover, it should be disabled after fixing the bug in Qt5.
https://bugreports.qt.io/browse/QTBUG-67023
https://bugreports.qt.io/browse/QTBUG-65682

nedrichards replied:
So those bugs will be fixed in Qt 5.11, right? That's due on the 31st May 2018.

scx replied:
Yes, it should be fixed in Qt 5.11. However, I have already provided a workaround.

nedrichards replied:
Sure, personally I'm much more fine with a hacky sort of workaround if it's going to be removed in the next month or so than I would be if those bugs hadn't been fixed yet.

nedrichards commented on README.md:
```diff
@@ -0,0 +1,124 @@
+* `"buildsystem": "simple"` and `build-commands: []`
+* `"buildsystem": "cmake-ninja"`
+
+It is possible to install flatpak-builder (CLI tool, IDE for GNOME in another thing) from flathub. However, it requires **flatpak** >= *0.10.0*
```
Yeah - the version that matters here is what flathub supports, whch is currently 0.11.x. Over time that will likely change, but for now being able to use the latest features and bug fixes is important.

scx replied:
I already tested it against:
- flatpak 0.8.7 and flatpak-builder 0.8.7 (EL7)
- flatpak 0.10.3 and flatpak-builder 0.10.10 (FC27)
- flatpak 0.10.4 and flatpak-builder 0.10.10 (FC27)
- flatpak 0.11.3 and flatpak-builder 0.10.10 (FC28)

nedrichards commented:
bot, build me.mitya57.ReText

nedrichards commented:
bot, build me.mitya57.ReText

TingPing commented on me.mitya57.ReText.json:
```diff
@@ -0,0 +1,382 @@
+                    size=\"${s}x${s}\";
+                    echo \"- ${size}\";
+                    mkdir -p \"icons/${size}/\";
+                    convert -background none -density 1024 -resize \"${size}\" \"${icon}\" \"icons/${size}/${appname}.png\";
```
As mentioned in the other one `rsvg-convert` is in the sdk so you can drop the build dep.

TingPing commented on me.mitya57.ReText.json:
```diff
@@ -0,0 +1,382 @@
+                done;",
+                "echo 'post-install: validate data';",
+                "cat /app/share/applications/*.desktop;",
+                "desktop-file-validate /app/share/applications/*.desktop || :;",
```
`desktop-file-install` already validates them.

scx commented:
I was unable to build PyQt >= 5.11. This is probably related to the changes in SIP.
https://stackoverflow.com/questions/51098152/installing-pyqt5-from-sources-unable-to-import-pyqt5-sip
http://pyqt.sourceforge.net/Docs/PyQt5/installation.html#downloading-sip
http://pyqt.sourceforge.net/Docs/sip4/using.html#ref-private-sip
http://pyqt.sourceforge.net/Docs/PyQt5/incompatibilities.html#importing-the-sip-module
Moreover, ReText crashes immediately after launch when using PyQt >= 5.10, so I have to stay with 5.9 branch. Unfortunately, it requires several patches to work with Qt >= 5.11.
I also had a problem with building PyQt on top of the SIP >= 4.19.9. In any case, the combination of Qt 5.11, SIP 4.19.8 and PyQt 5.9.2 (with patches) should be the best for ReText 7.0.3 at this moment.
I want to draw attention to the fact that the external web engine used by ReText was unstable when using Qt 5.9. This bug was repaired in Qt 5.11. That's why I chose KDE Platform/SDK 5.11.

BTW, I noticed that literally nobody uses PyQt 5.11. Something is definitely wrong with this version. Even Fedora Rawhide, Arch and Gentoo stayed with PyQt 5.10.
https://koji.fedoraproject.org/koji/packageinfo?packageID=17997
https://www.archlinux.org/packages/extra/x86_64/python-pyqt5/
https://packages.gentoo.org/packages/dev-python/PyQt5

nedrichards commented:
bot, build me.mitya57.ReText

nedrichards commented:
bot, build me.mitya57.ReText

nedrichards commented:
Builds clean: https://flathub.org/builds/#/builders/2/builds/2624

nedrichards commented:
Builds fine - https://flathub.org/builds/#/builders/2/builds/2624 I find the whole way that this is laid out to be somewhat confusing, but that's not a blocker unless someone who's actually reviewed it properly like @TingPing agrees. Could you look again in your own time?

scx commented:
> None of our major gripes seem actually fixed.

The main complaint was that I used extensive commands directly in the JSON manifest instead of in a separate script.
https://github.com/flathub/flathub/pull/376#discussion_r188920517
https://github.com/flathub/flathub/pull/376#pullrequestreview-120993839
I've already fixed that.

Now you want me to undo some of these changes, just to use the custom build system.
https://github.com/flathub/flathub/pull/394#discussion_r188921633
https://github.com/scx/flathub/blob/16b0584d95ee38e91d223334113bc800fd0a9a0b/boost.json#L88-L93
https://github.com/scx/flathub/blob/16b0584d95ee38e91d223334113bc800fd0a9a0b/boost-Makefile

> Still ships shell wrappers over common commands

I don't think that `strip-json-comments` belong to "common commands". Anyway, the main reason why I use these scripts is to remove comments from JSON files. I do not see any valid reason to remove these scripts from the repository.

Anyway, similar scripts are provided by the **Mark Text**, **Visual Studio Code**, **QuodLibet**, **BetaflightConfigurator** and even **GNOME Devhelp**, but I have not seen any comments on this.
https://github.com/flathub/com.github.marktext.marktext/tree/3084967fc2d6068ff4ed0118c0558973e3ed9365#scripts
https://github.com/flathub/com.visualstudio.code.oss/tree/722799c1e21f13d9ba1de4aa019c58bc70e4c2e0
https://github.com/flathub/io.github.quodlibet.QuodLibet/tree/0e1b8215ba55499b5889b903c4649aa5e512f1d7#flatpak-packaging-for-quod-libet
https://github.com/flathub/io.github.betaflight.BetaflightConfigurator/tree/62c8ea8794d296e2c386b7792f88bb7e97c80d44
https://github.com/flathub/org.gnome.Devhelp/tree/7765c77ee1ae22faebcf23fd832420fc48d49742

> overly lenghty readme

Again, it doesn't hurt anyone.

> that duplicates official documentation

It's not true. Yes, it contains references to the flatpak setup, but also provides useful information about the project and potential problems.

Again, similar documentation is provided by **Qutebrowser**, **LBRY**, **PyCharm**, **HandBrake**, **Mark Text**, **RetroArch**, **Transmission**, **QuodLibet**, **FreeFileSync**, **Quetoo**, etc. but I have not seen any comments on this.
https://github.com/glatzor/flathub/tree/a8dcfd403f5db94cef8e6a765397dfe8442dc12d#builing-the-flatpak
https://github.com/flathub/io.lbry.lbry-app/tree/586764dc1f47b9c07147b426b45157f4ca6d91b3#building
https://github.com/flathub/com.jetbrains.PyCharm-Community/tree/e3512ec9b802cf0298725f7cddfc46131d22e159#how-to-build-pycharm-community
https://github.com/flathub/fr.handbrake.ghb/tree/f5faf35312255faebee603bb1670362937dae887#flatpak-installation
https://github.com/flathub/com.github.marktext.marktext/tree/3084967fc2d6068ff4ed0118c0558973e3ed9365#scripts
https://github.com/flathub/org.libretro.RetroArch/tree/3619e81702a310831a54fb5eb061553feaf976d9#development
https://github.com/flathub/com.transmissionbt.Transmission/tree/cc04b14bbfee9a72d8dc01a3f31c63fb13d7f52f#building-and-testing-locally
https://github.com/flathub/io.github.quodlibet.QuodLibet/tree/0e1b8215ba55499b5889b903c4649aa5e512f1d7#flatpak-packaging-for-quod-libet
https://github.com/flathub/org.freefilesync.FreeFileSync/tree/d0e5f97b60b9de53143058387b090456d2db60ce#installation
https://github.com/flathub/org.quetoo.Quetoo/tree/93717008aec0ed74533680002c7b3ed1e7af1a72#build

What's more, a lot of applications provide completely worthless documentation, but nobody cares about this. For example: **MediaConch**, **AVIMetaEdit**, **QuasselClient**, **QCTools**, **MOVMetaEdit**, **DVAnalyzer**, **BWFMetaEdit**.
https://github.com/flathub/net.mediaarea.MediaConch/tree/267dbe6953f889d9665253c7edc0ec02627b203b#building-applications
https://github.com/flathub/net.mediaarea.AVIMetaEdit/tree/ef38615b89fffd38316e177b950c09704ff70922#building-applications
https://github.com/flathub/org.quassel_irc.QuasselClient/tree/5de06c3d78fad71c204219f6b24033505a031291#building-applications
https://github.com/flathub/net.mediaarea.QCTools/tree/06aaecb1e60d2641213883d1e3e222bf8b937d1c#building-applications
https://github.com/flathub/net.mediaarea.MOVMetaEdit/tree/3c6dfc21b4261d156ae5b420ef29f5bea8ce9b03#building-applications
https://github.com/flathub/net.mediaarea.DVAnalyzer/tree/ebc458b999d217fe2945e1488b38f2906972dae4#building-applications
https://github.com/flathub/net.mediaarea.BWFMetaEdit/tree/67ffc1ac433b25250da016ee17b6d2a41ddcb96b#building-applications

Since you are already referring to the official documentation, I would like to remind you that it says that the officially supported and recommended version of the flatpak for RHEL is the one that comes from the official RHEL repository.
https://web.archive.org/web/20180704055618/https://flatpak.org/setup/Red%20Hat%20Enterprise%20Linux/
Currently, it is flatpak 0.8.8. If you think this is a bug, you can report it on Red Hat Bugzilla.

BTW, from your perspective, the Flatpak documentation is probably a bit outdated, isn't it?
https://web.archive.org/web/20180704064500/http://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html
https://web.archive.org/web/20180704064638/http://docs.flatpak.org/en/latest/flatpak-command-reference.html

> uses weird template files that aren't even generated with some tool, doesn't use pip generator

I used it before, but it generated problematic code, so I fixed files manually. Anyway, it is not required to use flatpak-builder-tools, and there is no official packaging guidelines (equivalent of the Fedora Packaging Guidelines) for Flathub.

By the way, the **Paperwork** on Flathub uses pip-Makefile and I think it's a better concept (except it uses `setup.py` instead of `pip3`), but I decided to give it up because of your aversion to Makefile files.
https://github.com/flathub/work.openpaper.Paperwork/blob/8ea2466ac39150de0d61177de6f9f49934e8cb50/work.openpaper.Paperwork.json#L32-L36
https://github.com/flathub/work.openpaper.Paperwork/blob/8ea2466ac39150de0d61177de6f9f49934e8cb50/pip-Makefile

I can provide my own generator of JSON manifests for PIP packages, but I'm afraid you wouldn't like it because of your aversion to bash scripts.

> doesn't use simple buildsystem

Which does not change anything about code clarity. Current JSON files have exactly the same level of code clarity as if they were using a simple build system.

Anyway, I do not see any valid reason to do this:
https://github.com/flathub/flathub/pull/394#discussion_r188921633
https://github.com/scx/flathub/blob/16b0584d95ee38e91d223334113bc800fd0a9a0b/boost.json#L88-L93
https://github.com/scx/flathub/blob/16b0584d95ee38e91d223334113bc800fd0a9a0b/boost-Makefile

A lot of applications on Flathub use custom Makefiles and nobody has complained about it. Personally, I think this is a much better approach than the thoughtless use of the simple build system. I think that even @TingPing will agree with me on this matter.
https://github.com/flathub/work.openpaper.Paperwork/blob/8ea2466ac39150de0d61177de6f9f49934e8cb50/work.openpaper.Paperwork.json#L32-L36
https://github.com/flathub/work.openpaper.Paperwork/blob/8ea2466ac39150de0d61177de6f9f49934e8cb50/pip-Makefile
https://github.com/flathub/org.frozen_bubble.frozen-bubble/blob/c0007352f28cb975f97dd3728fd9159bcb736a31/org.frozen_bubble.frozen-bubble.json#L81-L85
https://github.com/flathub/org.frozen_bubble.frozen-bubble/blob/c0007352f28cb975f97dd3728fd9159bcb736a31/perl-BuildPL-Makefile
https://github.com/flathub/net.blockout.BlockOutII/blob/894c5cc98e1d035abb2eb8cf9ff1539ab5373f40/net.blockout.BlockOutII.json#L172-L176
https://github.com/flathub/net.blockout.BlockOutII/blob/894c5cc98e1d035abb2eb8cf9ff1539ab5373f40/blockout2-Makefile
https://github.com/flathub/net.sourceforge.btanks/blob/d569d30ab67cd6dd63d643543ae9f763a8649660/net.sourceforge.btanks.json#L45-L49
https://github.com/flathub/net.sourceforge.btanks/blob/d569d30ab67cd6dd63d643543ae9f763a8649660/scons-Makefile
https://github.com/flathub/com.teeworlds.Teeworlds/blob/001b538716b67dfdbe0b0071c386fc2b196a7df1/com.teeworlds.Teeworlds.json#L76-L80
https://github.com/flathub/com.teeworlds.Teeworlds/blob/001b538716b67dfdbe0b0071c386fc2b196a7df1/teeworlds-Makefile

I really don't see the point of using simple build system, when it does not make sense or does not bring any benefits.
Please look at the manifest of the **CuteMarkEd**.
https://github.com/flathub/io.github.cloose.CuteMarkEd/blob/81f4aa35482b15ecd2160e7280e5c95eb5c1a7d0/io.github.cloose.CuteMarkEd.json#L41-L62
It uses a simple build system instead of qmake.
https://github.com/scx/io.github.cloose.CuteMarkEd/blob/c8e6d705253cfb97a3441b592b322c1224afa03d/io.github.cloose.CuteMarkEd.json#L41-L75
Even autotools would be better in this case.
https://github.com/scx/io.github.cloose.CuteMarkEd/blob/bad4a65535e34444d9137ec2451212169ded16c6/io.github.cloose.CuteMarkEd.json#L41-L76
Not only this code is more readable, but it also brings benefits in the form of faster compilation (thanks to the `-j` parameter in `make`).

Please keep in mind that the officially supported and recommended (by both Red Hat and Flatpak/Flathub) version of flatpak for RHEL does not support simple build system, and I don't intend to use unofficial packages from an AMD64-only test repository.
As I said before, if you think this is a bug, you are free to report it on Red Hat Bugzilla.
https://bugzilla.redhat.com/show_bug.cgi?id=1570030

Just to be clear, I also tested all my Flatpak apps with the latest flatpak/flatpak-builder (currently 0.99.2) on Fedora 28. I have no problems with that.

> installs man pages that flatpak doesn't export

I know that currently flatpak does not export man files. However, this can change in the future and I try to create packages looking a little further ahead. Anyway, I see no reason to remove this, since it doesn't spoil anything in any way.

> and bundles qtwebkit built against 5.11 instead of using baseapp for 5.9.

I have discussed this issue with @nedrichards over a month ago. There were absolutely no objections to use Qt 5.11.
https://github.com/flathub/flathub/pull/376#discussion_r186347263
You were perfectly aware of the problem with stability of Qt 5.9, but during this time you did literally nothing to backports patches from Qt 5.11 to Qt 5.9. What do you expect from me? That I will use the broken runtime, because you care more about the misunderstood code clarity or build time than the stability and quality of the application?
Of course, I would be happy to use BaseApp for Qt5 Webkit (like `io.qt.qtwebkit.BaseApp`), but you don't provide this for Qt 5.11 runtime (`org.kde.Platform//5.11`).

> It's not some personal vendetta.

Yeah, it is pure immaturity and oversized ego problem.

> Clarity of what's going on is important, especially on Flathub, which we want to be seen as source of good practices.

I can point to dozens of applications on Flathub, which are much worse quality than this, both in terms of stability and code clarity of JSON files.
I bet that both @nedrichards and @alexlarsson will agree with me that the current JSON files are clarity enough to meet Flathub standards.

It is clear to me that you care more about misunderstood code clarity than the quality and stability of Flathub apps. In fact, your choices are based on personal preferences instead of objective criteria. That is why we have here a super unstable apps (like PiTiVi) and a lot of very poor games, already abandoned by their authors years ago.
What's worse, almost every major application here that is using WebKit/Chromium-based web engine has some issues, including stability problems. This is due to the lack of support for nested sandboxes and poor quality of the provided runtimes (and Qt web engines themselves, but this is another story).
https://lists.freedesktop.org/archives/flatpak/2017-June/000722.html
https://github.com/lirios/browser/issues/77
https://github.com/lirios/browser/issues/77#issuecomment-346407574
https://github.com/flathub/flathub/pull/288#discussion_r171910092
https://github.com/flathub/org.subsurface_divelog.Subsurface/blob/cf66cc179a574810f8d0c0353a3a2ffc47136679/org.subsurface_divelog.Subsurface.json#L19
https://github.com/glatzor/flathub/blob/a8dcfd403f5db94cef8e6a765397dfe8442dc12d/org.qutebrowser.qutebrowser.json#L10
https://github.com/jurf/flathub/blob/2613d12e7d418fd78472ea2da995634a7b44ba09/org.otter_browser.OtterBrowser.json#L19

This applies not only to web engines but even to glibc from Freedesktop. And as we can see, nobody really cares about it.
https://github.com/flatpak/freedesktop-sdk-images/issues/97#issuecomment-383414574
https://github.com/flatpak/freedesktop-sdk-images/issues/97#issuecomment-392346027

To summarize:
1. You don't like the fact that I provide documentation that contains instructions for building a flatpak package as well as a description of known issues. However, you don't have similar objections to the apps that are already on Flathub (or will be soon), such as the **Qutebrowser**, **RetroArch**, **RetroArch**, **FreeFileSync**, etc.
https://github.com/glatzor/flathub/tree/a8dcfd403f5db94cef8e6a765397dfe8442dc12d#builing-the-flatpak
https://github.com/flathub/com.github.marktext.marktext/tree/3084967fc2d6068ff4ed0118c0558973e3ed9365#scripts
https://github.com/flathub/org.libretro.RetroArch/tree/3619e81702a310831a54fb5eb061553feaf976d9#development
https://github.com/flathub/org.freefilesync.FreeFileSync/tree/d0e5f97b60b9de53143058387b090456d2db60ce#installation

2. You don't like my helper scripts, but you don't mind similar scripts in **Mark Text** or even **GNOME Devhelp**.
https://github.com/flathub/com.github.marktext.marktext/tree/3084967fc2d6068ff4ed0118c0558973e3ed9365#scripts
https://github.com/flathub/org.gnome.Devhelp/tree/7765c77ee1ae22faebcf23fd832420fc48d49742
https://github.com/flathub/org.gnome.Devhelp/blob/7765c77ee1ae22faebcf23fd832420fc48d49742/4-run-app.sh

3. You want to force me to use the simple build system instead of custom Makefile, even if it doesn't make sense or doesn't bring any benefits.
https://github.com/flathub/flathub/pull/394#discussion_r188921633
https://github.com/scx/flathub/blob/16b0584d95ee38e91d223334113bc800fd0a9a0b/boost.json#L88-L93
https://github.com/scx/flathub/blob/16b0584d95ee38e91d223334113bc800fd0a9a0b/boost-Makefile
However, you don't mind that others (e.g. **Frozen Bubble**) use custom Makefiles.
https://github.com/flathub/org.frozen_bubble.frozen-bubble/blob/c0007352f28cb975f97dd3728fd9159bcb736a31/org.frozen_bubble.frozen-bubble.json#L98-L102
https://github.com/flathub/org.frozen_bubble.frozen-bubble/blob/c0007352f28cb975f97dd3728fd9159bcb736a31/perl-MakefilePL-Makefile

4. You want to force me to use the flatpak-pip-generator. However, you don't mind that others (e.g. **Paperwork**, **KDE Connect Indicator**, **Flowblade**, **Anki**, **GNOME Builder**, **GNOME Eolie**, **GNOME Lollypop**, **GNOME Meld**, **GNOME Music**, **gPodder**, **OpenShot**, **Pitivi**, **Zim**) take a different approach.
https://github.com/flathub/work.openpaper.Paperwork/blob/8ea2466ac39150de0d61177de6f9f49934e8cb50/work.openpaper.Paperwork.json#L32-L36
https://github.com/flathub/work.openpaper.Paperwork/blob/8ea2466ac39150de0d61177de6f9f49934e8cb50/pip-Makefile
https://github.com/flathub/com.github.bajoja.indicator-kdeconnect/blob/f9b82a3d112c21b82345ffb1535d099b58d98b01/com.github.bajoja.indicator-kdeconnect.json#L82-L123
https://github.com/flathub/io.github.jliljebl.Flowblade/blob/8270cb446c3540745e5aae3da4c6bcc9f4649294/io.github.jliljebl.Flowblade.yaml#L160-L168
https://github.com/flathub/net.ankiweb.Anki/blob/b62bc0fea594df26fafd112d7d809cdb8782fa91/markdown.json#L1-L14
https://github.com/flathub/net.ankiweb.Anki/blob/b62bc0fea594df26fafd112d7d809cdb8782fa91/send2trash.json#L1-L14
https://github.com/flathub/net.ankiweb.Anki/blob/b62bc0fea594df26fafd112d7d809cdb8782fa91/pyaudio.json#L1-L14
https://github.com/flathub/net.ankiweb.Anki/blob/b62bc0fea594df26fafd112d7d809cdb8782fa91/decorator.json#L1-L14
https://github.com/flathub/net.ankiweb.Anki/blob/b62bc0fea594df26fafd112d7d809cdb8782fa91/requests.json#L1-L34
https://github.com/flathub/net.ankiweb.Anki/blob/b62bc0fea594df26fafd112d7d809cdb8782fa91/bs4.json#L1-L19
https://github.com/flathub/org.gnome.Builder/blob/e02140d2d7e154acdd92e0fcc401746f2b6a73a7/org.gnome.Builder.json#L177-L302
https://github.com/flathub/org.gnome.Eolie/blob/2fc68bf47c607bb1dff9449a03698361e32917ed/org.gnome.Eolie.json#L92-L243
https://github.com/flathub/org.gnome.Lollypop/blob/19e7c80de8ba4dcb7b7b9c1c6b4092437a323e57/org.gnome.Lollypop.json#L60-L209
https://github.com/flathub/org.gnome.meld/blob/bbd58f5a5c9d08e2b7d8fe6623ff85b9de51c949/org.gnome.meld.json#L55-L66
https://github.com/flathub/org.gnome.Music/blob/d3d7156c08b37a69d783527d38bf94e66014e131/org.gnome.Music.json#L122-L181
https://github.com/flathub/org.gpodder.gpodder/blob/21dd465322e4853f6df633903d309897aedc10ef/org.gpodder.gpodder.json#L24-L51
https://github.com/flathub/org.openshot.OpenShot/blob/578cc5108c92c4ac1c235e7b2eafc038aa574ee4/org.openshot.OpenShot.yaml#L103-L167
https://github.com/flathub/org.pitivi.Pitivi/blob/bdaf0c7d12640f70aa250f011e6887e88b21cdad/org.pitivi.Pitivi.json#L62-L136
https://github.com/flathub/org.zim_wiki.Zim/blob/da1eb724b5092d0f7299cb89666c2e4df6e516ea/org.zim_wiki.Zim.json#L65-L78

5. The worst part, however, is that you want to force me to use a broken and buggy Runtime/BaseApp, that causes application instability. And for what? Just to save time for building WebKit. This is totally insane. I see no reason why I should use `org.kde.Platform//5.9` where the application is unstable or at least feature-limited, when I can use `org.kde.Platform//5.11` where everything works just fine.
https://github.com/flathub/flathub/pull/376#discussion_r186347263

If you think that the quality of my package is too high for Flathub, you can close this request because I am not going to lower it because of your whims. First, however, I would like @nedrichards and @alexlarsson to speak on this topic, because I consider your objections to be pure personal.

barthalion commented:
Rules are fluid and just because something has been accepted before doesn't mean it would be now. Good that you know how to use grep but no fingerpointing at other applications, whether their manifests (that should be fixed, obviously patches welcome) or stability (report issues upstream) is going to change what I said here.

Also I'm not sure if you are just having fun or simply trolling by your ad hominem but I'll just let it go.

Things to fix:
[] - use `simple` build system instead of Makefile hackery
[] - stop using templates obfuscating what's going on
[] - stop shipping shell wrappers that are not simplifying anything

This is the bare minimum to accept your manifest on Flathub.

scx commented:
With Qt 5.11 stability has improved.

The only remaining problem is related to a bug in glibc. 

```
Inconsistency detected by ld.so: dl-close.c: 811: _dl_close: Assertion `map->l_init_called' failed!
```

However, it is not critical. This will be fixed in KDE 5.12 runtime.

barthalion commented:
Have you contacted upstream about your pull request?

barthalion commented:
bot, build me.mitya57.ReText

scx commented:
@barthalion 

> Have you contacted upstream about your pull request?

Yes.
https://github.com/retext-project/retext/issues/412

scx commented on me.mitya57.ReText.yaml:
```diff
@@ -0,0 +1,54 @@
+  # for access images
+  - --filesystem=host
+  # Network access
+  # WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-FmxTjHhxVf: Connection refused
```
> Does it actually affect anything?

It should be obvious. This is the Markdown editor, so you can refer to both local and remote images.

Example:
```
![Image of Yaktocat](https://octodex.github.com/images/yaktocat.png)
```

![Image of Yaktocat](https://octodex.github.com/images/yaktocat.png)

nedrichards commented:
bot, build me.mitya57.ReText

scx commented:
[AppData]: a pull request has been sent to the upstream vendor.
https://github.com/retext-project/retext/pull/426

scx commented:
@barthalion @TingPing @nedrichards @alexlarsson 
Could you tell me why you are still blocking the release of this application? I fail to see any technical obstacles to this. All issues have been solved a long time ago.

scx commented:
testbot, build me.mitya57.ReText

scx commented:
bot, build me.mitya57.ReText

scx commented:
bot, build me.mitya57.ReText

scx commented:
bot, build me.mitya57.ReText

felipehw commented:
@scx Why this Flatpak didn't happen?
